package/MDAnalysis/analysis/backends.py

```diff
@@ -0,0 +1,296 @@
+"""Analysis backends --- :mod:`MDAnalysis.analysis.backends`
+============================================================
+
+Provides :class:`BackendBase` base class to implement custom
+backends for :meth:`MDAnalysis.analysis.base.AnalysisBase.run()` and its
+subclasses. Two built-in backend classes classes are provided:
+:class:`BackendMultiprocessing` that supports parallelization via standard
+python ``multiprocessing`` module, and :class:`BackendDask`, that uses the same
+process-based parallelization as :class:`BackendMultiprocessing`, but different
+serialization algorithm (see https://distributed.dask.org/en/latest/serialization.html
+for explanation of ``dask`` serialization algorithms; ``multiprocessing`` uses
+default ``pickle`` serialization).
+"""
+import warnings
+from typing import Callable
+from MDAnalysis.lib.util import is_installed
+
+
+class BackendBase:
+    """Base class for backend implementation. Initializes an instance and performs
+    checks for its validity, such as n_workers and possibly other ones.
+
+    Parameters
+    ----------
+    n_workers : int
+        number of workers (usually, processes) over which the work is split
+
+    Examples
+    --------
+    .. code-block:: python
+    
+        from MDAnalysis.analysis.backends import BackendBase
+        class ThreadsBackend(BackendBase):
+            def apply(self, func, computations):
+                from multiprocessing.dummy import Pool
+
+                with Pool(processes=self.n_workers) as pool:
+                    results = pool.map(func, computations)
+                return results
+        from MDAnalysis.analysis.rms import RMSD
+        R = RMSD(...) # setup the run
+        n_workers = 2
+        backend = ThreadsBackend(n_workers=n_workers)
+        R.run(backend=backend, unsupported_backend=True)
+
+    .. versionadded:: 2.8.0
```

Comment:
Please check that the formatting in the docs is correct, eg https://mdanalysis--4162.org.readthedocs.build/en/4162/documentation_pages/analysis/backends.html has the versionchanged garbled. Basically, look at the generated pages and fix any reST that displays funnily. Possibly use two empty lines before versionchanged

Reply:
I (think) it's fine now, at least from what I can see in my sections.

But if you search "versionchanged" or "versionadded" in documentation, you still have few hits in both -- but I guess it shouldn't be this way